src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java

```diff
@@ -0,0 +1,110 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.task;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
+
+import net.fabricmc.loom.LoomGradleExtension;
+import net.fabricmc.loom.configuration.providers.mappings.MappingsProviderImpl;
+import net.fabricmc.loom.task.service.TinyRemapperBuildService;
+import net.fabricmc.loom.util.Constants;
+import net.fabricmc.loom.util.TinyRemapperHelper;
+import net.fabricmc.tinyremapper.TinyUtils;
+import net.fabricmc.tinyremapper.extension.mixin.MixinExtension;
+
+public class RemapTaskConfiguration {
+	private static final String REMAP_JAR_TASK_NAME = "remapJar";
+
+	public static void setupRemap(Project project) {
+		TaskContainer tasks = project.getTasks();
+
+		// Configure the default jar task
+		tasks.named(JavaPlugin.JAR_TASK_NAME, AbstractArchiveTask.class).configure(task -> {
+			task.getArchiveClassifier().set("dev");
+			task.finalizedBy(tasks.getByName(REMAP_JAR_TASK_NAME));
+		});
+
+		// Register the default remap jar task
+		tasks.register(REMAP_JAR_TASK_NAME, RemapJarTask.class, task -> {
+			final AbstractArchiveTask jarTask = tasks.named(JavaPlugin.JAR_TASK_NAME, AbstractArchiveTask.class).get();
+
+			// Basic task setup
+			task.dependsOn(jarTask);
+			task.setDescription("Remaps the built project jar to intermediary mappings.");
+			task.setGroup(Constants.TaskGroup.FABRIC);
+			project.getArtifacts().add("archives", task);
+
+			// Setup the input file and the nested deps
+			task.getInputFile().convention(jarTask.getArchiveFile());
+			task.getNestedJars().plus(project.getConfigurations().getByName(Constants.Configurations.INCLUDE));
+
+			// Setup the remapper service
+			final FileCollection classpath = project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME);
+
+			task.getTinyRemapperBuildService().convention(getOrCreatetinyRemapperBuildServiceProvider(project,
+					task.getSourceNamespace(),
+					task.getTargetNamespace(),
+					classpath
+			));
+		});
+	}
+
+	private static Provider<TinyRemapperBuildService> getOrCreatetinyRemapperBuildServiceProvider(final Project project, final Provider<String> from, final Provider<String> to, final FileCollection classpath) {
```

Comment:
Minor nitpick: `Tiny` should be capitalised

Reply:
Fixed.